tee-worker/omni-executor/rpc-server/src/methods/pumpx/submit_swap_order.rs

```diff
@@ -0,0 +1,214 @@
+use crate::{
+	error_code::*, oneshot, server::RpcContext, verify_auth::verify_auth_token_authentication,
+	Decode, Deserialize, ErrorCode,
+};
+use executor_core::native_task::*;
+use executor_primitives::{BoundedVec, OmniAuth};
+use executor_storage::{PumpxJwtStorage, Storage};
+use heima_authentication::auth_token::AUTH_TOKEN_ACCESS_TYPE;
+use heima_hex_utils::decode_hex;
+use heima_primitives::{
+	Address20, Address32, BinanceConfig, ChainAsset, CrossChainSwapProvider, EthereumToken,
+	Identity, Intent, PumpxConfig, PumpxOrderType, SingleChainSwapProvider, SolanaToken, SwapOrder,
+	Web2IdentityType,
+};
+use jsonrpsee::RpcModule;
+use native_task_handler::NativeTaskResponse;
+use pumpx::types::SwapType;
+
+// TODO: move this to a central place
+const SOLANA_CHAIN_ID: u32 = 10000;
+const ETHEREUM_CHAIN_ID: u32 = 1;
+const BSC_CHAIN_ID: u32 = 56;
+const BASE_CHAIN_ID: u32 = 8453;
+
+#[derive(Debug, Deserialize)]
+pub struct SubmitSwapOrderParams {
+	pub user_email: String,
+	pub order_type: PumpxOrderType,
+	pub swap_type: SwapType,
+	pub from_chain_id: u32,
+	pub from_token_ca: Option<String>,
+	pub from_amount: u64,
+	pub to_chain_id: u32,
+	pub to_token_ca: Option<String>,
+	pub double_out: bool,
+	pub is_one_click: bool,
+	pub token_cap: Option<String>,
+	pub price_usd: Option<String>,
+	pub trailing_percent: Option<u32>,
+	pub wallet_index: u32,
+	pub auth_token: String,
+}
+
+impl SubmitSwapOrderParams {
+	pub fn try_get_from_chain_asset(&self) -> Result<ChainAsset, ()> {
+		let from_chain_id = self.from_chain_id;
+		let from_token_ca = self.from_token_ca.clone();
+		Self::try_get_chain_asset(from_chain_id, from_token_ca)
+	}
+
+	pub fn try_get_to_chain_asset(&self) -> Result<ChainAsset, ()> {
+		let to_chain_id = self.to_chain_id;
+		let to_token_ca = self.to_token_ca.clone();
+		Self::try_get_chain_asset(to_chain_id, to_token_ca)
+	}
+
+	fn try_get_chain_asset(chain_id: u32, token_ca: Option<String>) -> Result<ChainAsset, ()> {
+		match chain_id {
+			SOLANA_CHAIN_ID => {
+				let solana_token = match token_ca {
+					None => SolanaToken::Native,
+					Some(ref s) if s.is_empty() => SolanaToken::Native,
+					Some(ref mint_address) => {
+						let mint_address: Address32 = mint_address.as_str().try_into()?;
```

Comment:
From their API example, they seem to pass in solana address format (base58), like
```
  "token_ca": "2qEHjDLDLbuBgRYvsxhc5D6uDWAivNFZGan56P1tpump",
```

So we need extra steps to convert it to Address32